src/status_im/chat/subs.cljs

```diff
@@ -219,8 +235,10 @@
   (fn [db [_ message-id]]
     (get-in db [:message-data :preview message-id :markup])))
 
-(reg-sub :get-message-preview
-  (fn [db [_ message-id]]
-    (let [preview (subscribe [:get-message-preview-markup message-id])]
-      (when-let [markup @preview]
-        (commands-utils/generate-hiccup markup)))))
+(reg-sub
+ :get-message-preview
+ (fn [[_ message-id]]
+   [(subscribe [:get-message-preview-markup message-id])])
+ (fn [[markup]]
```

Comment:
Why is this two fns now? And why is deref gone?

Reply:
we can't use `:<-` sugar here because of `message-id`, and i suppose `signal-fn` is kind of more "idiomatic" way to use `reg-sub` in this case. Because `computation-fn` is pure now.

https://github.com/Day8/re-frame/blob/master/src/re_frame/subs.cljc#L214

Reply:
actually i touched this code because i was under impression that subscription doesn't work properly because of derefs inside.